pkgs/development/interpreters/nextflow/default.nix

```diff
@@ -1,54 +1,102 @@
-{ lib
-, stdenv
-, fetchurl
-, makeWrapper
-, openjdk17
-, wget
-, which
-, gnused
-, gawk
-, coreutils
-, buildFHSEnv
+{
+  lib,
+  stdenv,
+  fetchurl,
+  makeWrapper,
+  openjdk,
+  gradle,
+  wget,
+  which,
+  gnused,
+  gawk,
+  coreutils,
+  testers,
+  nextflow,
+  bash,
 }:
-
-let
-  nextflow =
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: rec {
   pname = "nextflow";
-  version = "22.10.6";
+  version = "24.08.0-edge";
 
   src = fetchurl {
-    url = "https://github.com/nextflow-io/nextflow/releases/download/v${version}/nextflow-${version}-all";
-    hash = "sha256-zeYsKxWRnzr0W6CD+yjoAXwCN/AbN5P4HhH1oftnrjY=";
+    url = "https://github.com/nextflow-io/nextflow/archive/6e866ae81ff3bf8a9729e9dbaa9dd89afcb81a4b.tar.gz";
+    hash = "sha256-j1HFHn8Vu1/HLC1aAruFV7wm3otze8fTegAcf+cLrY4=";
   };
 
   nativeBuildInputs = [
     makeWrapper
-    openjdk17
+    gradle
+    bash
+    openjdk
     wget
     which
     gnused
     gawk
     coreutils
   ];
 
-  dontUnpack = true;
+  mitmCache = gradle.fetchDeps {
+    inherit (finalAttrs) pname;
+    data = ./deps.json;
+  };
+  __darwinAllowLocalNetworking = true;
+
+  patches = [ ./add-build-dependencies.patch ];
```

Comment:
I'm unhappy with this but could not get nextflow to compile otherwise. I don't know whether this is an issue with `mitm-cache` (i.e. maybe it does not correctly provide the dependencies) or an issue with nextflow's gradle config (i.e. it does not correctly list [transitive?] deps).

I was able to compile nextflow without issues in a `nix-shell` env with just `javac` & `make` at hand. But using the `mitm-cache` approach (as necessary for a derivation which should build deterministically), some deps were simply not fetched. For example, when removing `api "org.ow2.asm:asm-util:9.7"` (from the patch), the compilation aborts with

```bash
...nextflow> > Task :plugins:nf-wave:processResources
nextflow> > Task :nf-commons:compileGroovy FAILED
nextflow> > Task :plugins:nf-azure:copyPluginLibs
nextflow> > Task :plugins:nf-amazon:copyPluginLibs
nextflow> > Task :plugins:nf-google:copyPluginLibs
nextflow> FAILURE: Build failed with an exception.
nextflow> * What went wrong:
nextflow> Execution failed for task ':nf-commons:compileGroovy'.
nextflow> > Could not resolve all files for configuration ':nf-commons:detachedConfiguration1'.
nextflow>    > Could not find org.ow2.asm:asm-util:9.7.
nextflow>      Searched in the following locations:
nextflow>        - https://repo.maven.apache.org/maven2/org/ow2/asm/asm-util/9.7/asm-util-9.7.pom
nextflow>        - https://repo.eclipse.org/content/groups/releases/org/ow2/asm/asm-util/9.7/asm-util-9.7.pom
nextflow>        - https://oss.sonatype.org/content/repositories/snapshots/org/ow2/asm/asm-util/9.7/asm-util-9.7.pom
nextflow>        - https://s3-eu-west-1.amazonaws.com/maven.seqera.io/releases/org/ow2/asm/asm-util/9.7/asm-util-9.7.pom
nextflow>        - https://s3-eu-west-1.amazonaws.com/maven.seqera.io/snapshots/org/ow2/asm/asm-util/9.7/asm-util-9.7.pom
nextflow>      Required by:
nextflow>          project :nf-commons > org.apache.groovy:groovy-console:4.0.22
nextflow> * Try:
nextflow> > Run with --stacktrace option to get the stack trace.
nextflow> > Run with --info or --debug option to get more log output.
nextflow> > Run with --scan to get full insights.
nextflow> > Get more help at https://help.gradle.org.
nextflow> Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
nextflow> You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
nextflow> For more on this, please refer to https://docs.gradle.org/8.8/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
nextflow> BUILD FAILED in 40s
nextflow> 19 actionable tasks: 19 executed
```

I tested with

```bash
nix build -L .#nextflow.mitmCache.updateScript && ./result && nix build -L  .#nextflow
```

The `mitmCache.updateScript` runs throught but building itself yields the above errors.

Here is the diff to reproduce the error:

```diff
diff --git a/pkgs/development/interpreters/nextflow/add-build-dependencies.patch b/pkgs/development/interpreters/nextflow/add-build-dependencies.patch
index ccc774245d8c..cfd4d3629c16 100644
--- a/pkgs/development/interpreters/nextflow/add-build-dependencies.patch
+++ b/pkgs/development/interpreters/nextflow/add-build-dependencies.patch
@@ -2,7 +2,7 @@ diff --git a/modules/nf-commons/build.gradle b/modules/nf-commons/build.gradle
 index 2a5c6ec70..a29305106 100644
 --- a/modules/nf-commons/build.gradle
 +++ b/modules/nf-commons/build.gradle
-@@ -26,6 +26,12 @@ sourceSets {
+@@ -26,6 +26,11 @@ sourceSets {

  dependencies {
      api "ch.qos.logback:logback-classic:1.4.14"
@@ -10,7 +10,6 @@ index 2a5c6ec70..a29305106 100644
 +    api "org.apache.groovy:groovy-datetime:4.0.22"
 +    api "org.apache.groovy:groovy-dateutil:4.0.22"
 +    api "org.apache.groovy:groovy-sql:4.0.22"
-+    api "org.ow2.asm:asm-util:9.7"
 +    api "org.abego.treelayout:org.abego.treelayout.core:1.0.3"
      api "org.apache.groovy:groovy:4.0.22"
      api "org.apache.groovy:groovy-nio:4.0.22"
```

I basically just copy & pasted the "missing" deps from the error messages and build the patch accordingly.